visualization/app/codeCharta/util/treeMapGenerator.ts

```diff
@@ -12,7 +12,20 @@ export function createTreemapNodes(map: CodeMapNode, state: State, metricData: N
 	const heightScale = (state.treeMap.mapSize * 2) / maxHeight
 
 	if (hasFixedFolders(map)) {
-		return buildSquarifiedTreeMapsForFixedFolders(map, state, heightScale, maxHeight, isDeltaState)
+		const hierarchyNode = hierarchy(map)
+
+		// Base root folder has width: 100px and length: 100px
+		const nodes: Node[] = [TreeMapHelper.buildRootFolderForFixedFolders(hierarchyNode.data, heightScale, state, isDeltaState)]
+
+		// Multiply mapSize of (default) 250px by 2 = 500px and add the total margin
+		// than divide through the root folder width and length to get a scale factor for calculation for all following nodes.
+		const scaleLength = (state.treeMap.mapSize * 2 + getEstimatedNodesPerSide(hierarchyNode) * state.dynamicSettings.margin) / nodes[0].width
+		const scaleWidth = (state.treeMap.mapSize * 2 + getEstimatedNodesPerSide(hierarchyNode) * state.dynamicSettings.margin) / nodes[0].length
```

Comment:
store (state.treeMap.mapSize * 2 + getEstimatedNodesPerSide(hierarchyNode) * state.dynamicSettings.margin)
```suggestion
		const scale = (state.treeMap.mapSize * 2 + getEstimatedNodesPerSide(hierarchyNode) * state.dynamicSettings.margin)
		const scaleLength =  scale/ nodes[0].width
		const scaleWidth = scale / nodes[0].length
```

Reply:
Nice suggestion. I will apply it, however, I will rename the new variable name.